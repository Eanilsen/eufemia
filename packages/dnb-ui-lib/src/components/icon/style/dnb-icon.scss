/*
* DNB icon
*
* ======================================================================== */

@import '../../../style/component/core-imports.scss';
@import './_icon.scss';
