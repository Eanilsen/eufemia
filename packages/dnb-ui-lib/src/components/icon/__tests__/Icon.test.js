/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../Icon'
// just to make sure we re-run the test in watch mode due to changes in this file
import '../style/dnb-icon.scss'
import { question } from './test-files'

const props = fakeProps(require.resolve('../Icon'), {
  optional: true
})
props.icon = question

describe('Icon component', () => {
  it('have to match snapshot', () => {
    const Comp = mount(<Component {...props} />)
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('has valid width and height prop', () => {
    const width = 200
    const height = 100
    const Comp = mount(
      <Component {...props} width={width} height={height} />
    )
    const elem = Comp.find('svg')
    expect(elem.exists()).toBe(true)
    expect(elem.props().width).toBe(width)
    expect(elem.props().height).toBe(height)
  })

  it('should validate with ARIA rules', async () => {
    const Comp = mount(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('Icon scss', () => {
  const scss = loadScss(require.resolve('../style/dnb-icon.scss'))
  it('have to match snapshot', () => {
    expect(scss).toMatchSnapshot()
  })
})
