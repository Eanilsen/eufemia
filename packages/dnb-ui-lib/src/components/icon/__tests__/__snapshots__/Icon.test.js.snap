// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component have to match snapshot 1`] = `
<Icon
  alt="question mark"
  attributes="attributes"
  class="class"
  className="className"
  color="color"
  height="height"
  icon={[Function]}
  modifier="modifier"
  size="size"
  title="title"
  width="width"
>
  <span
    alt="question mark"
    aria-label="question mark"
    attributes="attributes"
    className="dnb-icon dnb-icon--modifier class className"
    role="img"
    title="title"
  >
    <SvgComponent
      color="color"
    >
      <svg
        color="color"
        viewBox="0 0 16 16"
      >
        <g
          clipPath="url(#clip0)"
        >
          <path
            clipRule="evenodd"
            d="M4.352 5.1c0-2.05 1.55-3.6 3.6-3.6 2.052 0 3.601 1.55 3.601 3.6 0 2.051-1.55 3.6-3.6 3.6a.75.75 0 0 0-.75.75v2.666a.75.75 0 0 0 1.5 0V10.15c2.506-.348 4.35-2.425 4.35-5.049 0-2.879-2.22-5.1-5.1-5.1-2.88 0-5.1 2.221-5.1 5.1a.75.75 0 0 0 1.5 0zm3.6 10.898c.659 0 1.097-.439 1.097-1.097 0-.658-.438-1.096-1.096-1.096-.658 0-1.097.438-1.097 1.096 0 .658.439 1.097 1.097 1.097z"
            fillRule="evenodd"
          />
        </g>
        <defs>
          <clipPath
            id="clip0"
          >
            <path
              d="M0 0h16v16H0z"
            />
          </clipPath>
        </defs>
      </svg>
    </SvgComponent>
  </span>
</Icon>
`;

exports[`Icon scss have to match snapshot 1`] = `
"/*
* DNB icon
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * CSS Properties
 *
 */
:root {
  --font-family-default: 'Fedra Sans Std', sans-serif;
  --font-family-book: 'Fedra Sans Std', sans-serif;
  /* Do not use, use --font-weight-... */
  --font-family-demi: 'Fedra Sans Std', sans-serif;
  /* Do not use, use --font-weight-... */
  --font-family-medium: 'Fedra Sans Std', sans-serif;
  /* Do not use, use --font-weight-... */
  --font-weight-default: normal;
  --font-weight-book: normal;
  --font-weight-demi: 500;
  --font-weight-medium: 600;
  --font-size-small: 0.875rem;
  --font-size-default: 1rem;
  --font-size-x-default: 1.25rem;
  --font-size-medium: 1.5rem;
  --font-size-large: 2rem;
  --font-size-xx-large: 3rem;
  --color-sea-green-alt: #008484;
  --color-mint-green-50: #d2f0e9;
  --color-mint-green-25: #e9f8f4;
  --color-mint-green-12: #f4fbf9;
  --color-sea-green-alt-30: #b3dada;
  --color-signal-yellow-30: #ffffd7;
  --color-cherry-red: #e10076;
  --color-cherry-red-80: #fdebf4;
  --color-black: #000;
  --color-black-80: #333;
  --color-black-30: #b3b3b3;
  --color-white: #fff;
  --color-black-border: #cdcdcd;
  --color-black-background: #fafafa;
  --color-sea-green: #007272;
  --color-mint-green: #a5e1d2;
  --color-summer-green: #28b482;
  --color-emerald-green: #14555a;
  --color-ocean-green: #00343e;
  --color-signal-orange: #ff5400;
  --color-signal-yellow: #ffff7a;
  --color-indigo: #23195a;
  --color-violet: #6e2382;
  --color-sky-blue: #4bbed2; }

/*
 * Typography
 *
 * The font-family properties are defined in the properties.scss file, like:
 * --font-family-default:
 *
 */
.dnb-typo-std,
.dnb-typo-book {
  font-family: var(--font-family-default);
  font-weight: normal;
  font-style: normal; }

.dnb-typo-medium {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-medium);
  font-style: normal; }

.dnb-typo-demi {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-demi);
  font-style: normal; }

.dnb-typo-number--lining {
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\"; }

.dnb-typo-number--tabular {
  font-variant-numeric: tabular-nums;
  font-feature-settings: \\"lnum\\"; }

.dnb-typo-number--oldstyle {
  font-variant-numeric: oldstyle-nums;
  font-feature-settings: \\"onum\\"; }

.dnb-typo-number--proportional {
  font-variant-numeric: proportional-nums;
  font-feature-settings: \\"pnum\\"; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal;
  font-display: fallback; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.ttf\\") format(\\"truetype\\");
  font-weight: 500;
  font-style: normal;
  font-display: fallback; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.ttf\\") format(\\"truetype\\");
  font-weight: 600;
  font-style: normal;
  font-display: fallback; }

/*
 * Scopes
 *
 */
/*
* Icon component
*
*/
.dnb-icon {
  display: inline-block;
  vertical-align: baseline;
  font-size: inherit;
  line-height: 1;
  width: 1em;
  height: 1em; }
  .dnb-icon svg:not([fill]),
  .dnb-icon svg [fill] {
    fill: currentColor; }
  .dnb-icon svg [stroke] {
    stroke: currentColor; }
  .dnb-icon--small {
    width: 0.75rem;
    height: 0.75rem; }
  .dnb-icon--default {
    width: 1rem;
    height: 1rem; }
  .dnb-icon--medium {
    width: 1.5rem;
    height: 1.5rem; }
  .dnb-icon--large {
    width: 2rem;
    height: 2rem; }
  .dnb-icon--custom-size {
    width: auto;
    height: auto; }
  .dnb-icon svg:not([width]):not([height]),
  .dnb-icon img:not([width]):not([height]) {
    width: inherit;
    height: inherit; }
  .dnb-icon svg[width='100%'] {
    width: inherit; }
  .dnb-icon svg[height='100%'] {
    height: inherit; }
"
`;
